corsheaders/middleware.py

```diff
@@ -150,11 +148,12 @@ def regex_domain_match(self, origin):
             if re.match(domain_pattern, origin):
                 return origin
 
-    def origin_found_in_model(self, url):
+    def origin_found_in_model(self, origin):
         if conf.CORS_MODEL is None:
             return False
         model = apps.get_model(*conf.CORS_MODEL.split('.'))
-        return model.objects.filter(cors=url.netloc).exists()
+        url = urlparse(origin)
+        return model.objects.filter(cors="{scheme}://{host}".format(scheme=url.scheme, host=url.netloc)).exists()
```

Comment:
the built-in model sucks because it's poorly thought through, and I wish it wasn't in the library as I think most people don't use it. But we surely can't just change the code here to expect a different format of strings in the database.

Reply:
Great catch!